components/sceneutil/shadow.cpp

```diff
@@ -66,6 +66,8 @@ namespace SceneUtil
     void ShadowManager::disableShadowsForStateSet(osg::ref_ptr<osg::StateSet> stateset)
     {
         int numberOfShadowMapsPerLight = Settings::Manager::getInt("number of shadow maps", "Shadows");
+        numberOfShadowMapsPerLight = std::max(1, std::min(numberOfShadowMapsPerLight, 8));
```

Comment:
If we're putting a bound on this when *disabling* shadows, we should also do it when *enabling* shadows here: https://github.com/OpenMW/openmw/pull/2596/files#diff-ea4f77bf8ee802245edb11cf81aa0556R26